### PR TITLE
Filter unused preferences

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
The following PR does a little bit of refactoring around preferences, also starts to filter out the preferences that are not used in the mobile app, this is basically to avoid having data in the database that is never used to keep the database with only relevant data, specially as users may have thousands of preferences.

Most importantly this PR fixes an issue where the preferences were not correctly sync'd when some preferences were deleted.

#### Ticket Link
N/A

```release-note
NONE
```

### CODE REVIEW

**Review by avinashlng1080 — APPROVED** [review]

**Review by anurag6713 — APPROVED** [review]

File: app/constants/preferences.ts [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
+    NOTIFICATIONS: 'notifications',
+    SAVED_POST: 'flagged_post',
+    SIDEBAR_SETTINGS: 'sidebar_settings',
+    TEAMS_ORDER: 'teams_order',
+    THEME: 'theme',
+};
+
+const CATEGORIES: Record<string, string> = {
+    ...CATEGORIES_TO_KEEP,
+    FAVORITE_CHANNEL: 'favorite_channel',
```
Comment (larkox) [inline app/constants/preferences.ts:19]: Why favorite channel is not on the categories to keep?

File: app/database/operator/server_data_operator/handlers/user.ts [hunk 2]
```diff
@@ -33,32 +34,35 @@ const UserHandler = (superclass: any) => class extends superclass {
...
             );
-            return [];
+            return records;
         }
 
         // WE NEED TO SYNC THE PREFS FROM WHAT WE GOT AND WHAT WE HAVE
         const deleteValues: PreferenceModel[] = [];
         const stored = await this.database.get(PREFERENCE).query().fetch() as PreferenceModel[];
-        const preferenesMap = new Map(stored.map((p) => {
+        const storedPreferesMap = new Map(stored.map((p) => {
```
Comment (larkox) [inline app/database/operator/server_data_operator/handlers/user.ts:49]:
```suggestion
        const storedPreferencesMap = new Map(stored.map((p) => {
```

File: app/helpers/api/preference.ts [hunk 3]
```diff
@@ -40,3 +34,27 @@ export function getTeammateNameDisplaySetting(preferences: PreferenceType[] | Pr
...
+
+export function getSidebarPreferenceAsBool(preferences: Preference[], name: string, defaultValue = false) {
+    return getPreferenceAsBool(preferences, Preferences.CATEGORIES.SIDEBAR_SETTINGS, name, defaultValue);
+}
+
+export function filterPreferences(preferences?: PreferenceType[]) {
+    if (!preferences?.length) {
+        return preferences;
+    }
+    const categories = new Set(Object.values(CATEGORIES_TO_KEEP));
```
Comment (larkox) [inline app/helpers/api/preference.ts:58]: Do we need to create this set every time? Does it make sense to have it precalculated  (i.e. outside of the function)?

**Review by larkox — APPROVED** [review]
LGTM, just a few minor nits.

File: app/constants/preferences.ts (hunk 1 above)
Comment (enahum) [reply app/constants/preferences.ts:19]: cause is no longer being used.. categories now hold the favorites `category` on the sidebar. confusing I know

**Pull request merged**